Formula/podman.rb

```diff
@@ -1,97 +1,229 @@
-    # the `$` for Makefile syntax, so we need to double it as `$$BINDIR`.
-    ENV["HELPER_BINARIES_DIR"] = "$$BINDIR/../libexec/podman" if OS.linux?
-    system "make", "podman-remote"
     if OS.mac?
-      bin.install "bin/#{os}/podman" => "podman-remote"
+      ENV["CGO_ENABLED"] = "1"
+
+      system "make", "podman-remote"
+      bin.install "bin/darwin/podman" => "podman-remote"
       bin.install_symlink bin/"podman-remote" => "podman"
+
       system "make", "podman-mac-helper"
-      bin.install "bin/#{os}/podman-mac-helper" => "podman-mac-helper"
+      bin.install "bin/darwin/podman-mac-helper" => "podman-mac-helper"
+
+      resource("gvproxy").stage do
+        system "make", "gvproxy"
+        (libexec/"podman").install "bin/gvproxy"
+      end
+
+      system "make", "podman-remote-darwin-docs"
+      man1.install Dir["docs/build/remote/darwin/*.1"]
+
+      bash_completion.install "completions/bash/podman"
+      zsh_completion.install "completions/zsh/_podman"
+      fish_completion.install "completions/fish/podman.fish"
     else
-      bin.install "bin/podman-remote"
-    end
+      paths = Dir["**/*.go"].select do |file|
+        (buildpath/file).read.lines.grep(%r{/etc/containers/}).any?
+      end
+      inreplace paths, "/etc/containers/", etc/"containers/"
+
+      ENV.O0
+      ENV["PREFIX"] = prefix
+      ENV["HELPER_BINARIES_DIR"] = opt_libexec/"podman"
+
+      system "make"
+      system "make", "install.bin", "install.remote", "install.man", "install.completions"
+
+      (prefix/"etc/containers/policy.json").write <<~EOS
+        {"default":[{"type":"insecureAcceptAnything"}]}
+      EOS
+
+      (prefix/"etc/containers/storage.conf").write <<~EOS
+        [storage]
+        driver="overlay"
+      EOS
+
+      (prefix/"etc/containers/registries.conf").write <<~EOS
+        unqualified-search-registries=["docker.io"]
+      EOS
+
+      resource("catatonit").stage do
+        system "./autogen.sh"
+        system "./configure"
+        system "make"
+        mv "catatonit", libexec/"podman/"
+      end
 
-    resource("gvproxy").stage do
-      system "make", "gvproxy"
-      (libexec/"podman").install "bin/gvproxy"
+      resource("netavark").stage do
+        system "cargo", "install", *std_cargo_args
+        mv bin/"netavark", libexec/"podman/"
+      end
+
+      resource("aardvark-dns").stage do
+        system "cargo", "install", *std_cargo_args
+        mv bin/"aardvark-dns", libexec/"podman/"
+      end
     end
+  end
 
-    system "make", "podman-remote-#{os}-docs"
-    man1.install Dir["docs/build/remote/#{os}/*.1"]
+  def caveats
+    on_linux do
+      <<~EOS
+        You need "newuidmap" and "newgidmap" binaries installed system-wide
+        for rootless containers to work properly.
+      EOS
+    end
+  end
 
-    bash_completion.install "completions/bash/podman"
-    zsh_completion.install "completions/zsh/_podman"
-    fish_completion.install "completions/fish/podman.fish"
+  service do
```

Comment:
This service should be Linux only but there seems to be no support in brew for situations like that.